spec/controllers/guides_controller_spec.rb

```diff
@@ -67,4 +67,12 @@
       end
     end
   end
+
+  describe "a malicious user trying to initialise an unwanted Guide STI constant" do
```

Comment:
There's an eval on params??

On Wednesday, 27 April 2016, Henry <notifications@github.com> wrote:

> In spec/controllers/guides_controller_spec.rb
> <https://github.com/alphagov/service-manual-publisher/pull/238#discussion_r61254003>
> :
>
> > @@ -67,4 +67,12 @@
> >        end
> >      end
> >    end
> > +
> > +  describe "a malicious user trying to initialise an unwanted Guide STI constant" do
>
> I added this test to drive adding params[:type].presence_in(%w{
> GuideCommunity Point }) which is defending against someone trying to slip
> something into Guide.new(type: type) which, deep deep down, is
> essentially performing an eval on params 😱
>
> It's probably too paranoid - I kind of the like the paranoia here though.
> Evalling params scares me!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/alphagov/service-manual-publisher/pull/238/files/74498aec675689696dd1691fa96d1137d026a584#r61254003>
>